dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApiLiveTest.java

```diff
@@ -0,0 +1,128 @@
+import com.google.common.collect.Lists;
+import org.jclouds.dimensiondata.cloudcontrol.domain.CpuSpeed;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Disk;
+import org.jclouds.dimensiondata.cloudcontrol.domain.NIC;
+import org.jclouds.dimensiondata.cloudcontrol.domain.NetworkInfo;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Server;
+import org.jclouds.dimensiondata.cloudcontrol.domain.State;
+import org.jclouds.dimensiondata.cloudcontrol.domain.options.CloneServerOptions;
+import org.jclouds.dimensiondata.cloudcontrol.internal.BaseDimensionDataCloudControlApiLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static org.jclouds.dimensiondata.cloudcontrol.utils.DimensionDataCloudControlResponseUtils.waitForServerState;
+import static org.jclouds.dimensiondata.cloudcontrol.utils.DimensionDataCloudControlResponseUtils.waitForServerStatus;
+import static org.jclouds.dimensiondata.cloudcontrol.utils.DimensionDataCloudControlResponseUtils.waitForVmToolsRunning;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "live", testName = "ServerApiLiveTest", singleThreaded = true)
+public class ServerApiLiveTest extends BaseDimensionDataCloudControlApiLiveTest {
+
+   private String serverId;
+   private String cloneImageId;
+   private final String deployedServerName = ServerApiLiveTest.class.getSimpleName() + System.currentTimeMillis();
+
+   @Test(dependsOnMethods = "testDeployAndStartServer")
+   public void testListServers() {
+      List<Server> servers = api().listServers().concat().toList();
+      assertNotNull(servers);
+      boolean foundDeployedServer = false;
+      for (Server s : servers) {
+         assertNotNull(s);
+         if (s.name().equals(deployedServerName)) {
+            foundDeployedServer = true;
+         }
+      }
+      assertTrue(foundDeployedServer, "Did not find deployed server " + deployedServerName);
+   }
+
+   @Test
+   public void testDeployAndStartServer() {
+      Boolean started = Boolean.TRUE;
+      NetworkInfo networkInfo = NetworkInfo
+            .create(NETWORK_DOMAIN_ID, NIC.builder().vlanId(VLAN_ID).build(), Lists.<NIC>newArrayList());
+      List<Disk> disks = ImmutableList.of(Disk.builder().scsiId(0).speed("STANDARD").build());
+      serverId = api().deployServer(deployedServerName, IMAGE_ID, started, networkInfo, "P$$ssWwrrdGoDd!", disks, null);
+      assertNotNull(serverId);
+      waitForServerStatus(api(), serverId, true, true, 30 * 60 * 1000, "Error");
+      waitForServerState(api(), serverId, State.NORMAL, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testDeployAndStartServer")
+   public void testReconfigureServer() {
+      api().reconfigureServer(serverId, 4, CpuSpeed.HIGHPERFORMANCE.name(), 1);
+      waitForServerState(api(), serverId, State.PENDING_CHANGE, 30 * 60 * 1000, "Error");
```

Comment:
Is it possible that the reconfigure operation is fast enough that we don't actually see this state? Does it really make sense to wait for intermediate states?

Reply:
I was finding some inconsistencies when running the test class. The extra check for PENDING_CHANGE state meant that the test was correctly validated. I have run the tests again without this check and it seems to be fine. I will remove,